### PR TITLE
feat(liquidator): disable 1inch functionality, warn user if they try to use it

### LINKED ISSUE #2056: disable 1-inch integration in liquidator bot in favor of withdraw defense strategy
**Why**
The 1inch integration in liquidator bot allows the bot to hold collateral and efficiently turn it into synthetic on the open market at time of liquidation. After working through the withdraw defense strategy, certain problems with the implementation are coming to light, and I would recommend disabling it in the short term to allow for a refactor later.

The issues are quite subtle but on a conceptual level there are also incompatibilities with the withdraw defense strategy. 
- 1inch integration buys just enough tokens to liquidate position fully, defense strategy requires that you save some tokens for withdraw extension fund. meaning there are 2 different visions of how bot balance should be managed, as well as how positions should be liquidated. This means defense strategy cannot just run after 1inch calls, and vice versa.
- defense strategy triggers when synthetic token balance is insufficient to fully liquidate position, but with one inch integration synthetic balance may be 0 until funds are purchased, rendering defense strategy unable to calculate a liquidation size.
- @mrice32 pointed this out, but 1inch integration does not correctly calculate the amount of tokens to buy, even with the collateral available to buy synthetics to fully liquidate it may always partially liquidate.
- When liquidating multiple positions, the current tokensToLiquidate logic overestimates your synthetic balance causing a liquidation to error rather than partial liquidation.  This should be replaced by the defense strategy implementation which fixes this problem.

1inch integration currently is not used in production, and probably shouldn't be until we can address these technical issues and other design issues around how to more robustly add it. 

**How**
- Deprecate usage of 1inch by logging a warning or throwing an error if configuration would enable it
- Move current 1inch logic into a standalone function which would do the collateral to synthetic conversion when called on
- update tests
- Revisit the full integration when the team feels this functionality needs to be automated 

**Resolution**
- See that 1inch logic is moved out of the `liquidatePositions` function and into a standalone function.
- See that there is visibility to the end user if they want to enable this feature that its no longer available (if needed)

**Difficulty Score [1-10]**
- 4: ~2 hours

### PR DESCRIPTION
**Pull request by daywiss**
**Motivation**

#2056 

**Summary**

This change to disable 1-inch integration is not the cleanest, but IMO is the lowest risk way to do it while maintaining the old code in its current form. No new tests were needed and no existing tests were removed, some tests were slightly tweaked. 
Old logic is fully intact and can easily be enabled again or referenced during a refactor. Essentially the old function was renamed, and copied into a new `liquidatePositions` function with the 1inch logic removed. 

User is warned now if trying to enable 1inch through an `info` style log. If 1inch is enabled it will be ignored during bot operation.

Fixes #2056

### CODE REVIEW

File: packages/liquidator/src/liquidator.js [hunk 1]
```diff
@@ -148,6 +148,210 @@ class Liquidator {
       at: "Liquidator",
       message: "Checking for liquidatable positions and preforming liquidations"
     });
+    // If an override is provided, use that price. Else, get the latest price from the price feed.
+    const price = liquidatorOverridePrice
```
Comment (nicholaspai) [inline packages/liquidator/src/liquidator.js:152]: So is all of this just old functionality re-added?

File: packages/liquidator/src/liquidator.js [hunk 2]
```diff
@@ -148,6 +148,210 @@ class Liquidator {
       at: "Liquidator",
       message: "Checking for liquidatable positions and preforming liquidations"
     });
+    // If an override is provided, use that price. Else, get the latest price from the price feed.
```
Comment (daywiss) [inline packages/liquidator/src/liquidator.js:151]: This is the beginning of the new liquidatePositions function, which basically copied the old code but strips out 1inch

File: packages/liquidator/src/liquidator.js [hunk 3]
```diff
@@ -148,6 +148,210 @@ class Liquidator {
...
+        txnConfig,
+        liquidationResult: logResult
+      });
+    }
+  }
+
+  // Process positions the same way, but pull in tokens from 1inch if balance is not adequate.
+  // This is currently disabled until we can take a second look at it. Its easy to re-enable
+  // by adding this specific call to process positions in the liquidator/index.
+  async liquidatePositionsWithOneInch(maxTokensToLiquidateWei, liquidatorOverridePrice) {
```
Comment (daywiss) [inline packages/liquidator/src/liquidator.js:350]: this is the new function which is identical to the previous liquidatePositions function, just renamed.

Comment (nicholaspai) [inline packages/liquidator/src/liquidator.js:350]: This seems like a ton of re-used code. Any reason not to add duplicate code to an internal method that is called by both `liquidatePositions` and `liquidatePositionsWithOneInch`?

**Review by nicholaspai — APPROVED** [review]
I think this is a really clean way to disable 1inch functionality because it doesn't remove/add any tests. I'd like to see more code re-use but I'm sure there's a reason for your design choices. Great work LGTM

File: packages/liquidator/src/liquidator.js (hunk 3 above)
Comment (daywiss) [reply packages/liquidator/src/liquidator.js:350]: A couple reasons
- refactoring the code could be pretty time consuming, and this was meant to be a quick change to unblock the WDF task
- while ugly the change is really low risk while allowing the old behavior to be re-enabled with a single line change
- would like to take on refactor as a seperate issue, and also revisit the 1inch integration in general to see how to better incorporate

That said though I am open to suggestions on what to do here, I just want to also be able to get the withdraw strategy PR in a reasonable time

Comment (mrice32) [reply packages/liquidator/src/liquidator.js:350]: Personally, I think this is fine the way you have it. The only other way to do this quickly with no real refactoring would be to just remove the `liquidatePositionsWithOneInch` function entirely, which would also be fine IMO.

**Review by mrice32 — APPROVED** [review]
This LGTM once this thread: https://github.com/UMAprotocol/protocol/pull/2058/files#r498847594 comes to a resolution.

File: packages/liquidator/src/liquidator.js (hunk 3 above)
Comment (daywiss) [reply packages/liquidator/src/liquidator.js:350]: Yea that I am ok with that, it will clean things up a lot.

**Comment by daywiss** [comment]
FYI when I removed the 1inch function, eslint rules required I remove unused variables, so I ended up having to strip out more than the function, basically all references to 1 inch.

**Pull request merged**